modules/core/src/core/model.js

```diff
@@ -598,14 +598,8 @@ count: ${this.stats.profileFrameCount}`
       if (descriptor instanceof Attribute) {
         attribute = descriptor;
       } else if (descriptor instanceof Buffer) {
-        attribute =
-          attribute ||
-          new Attribute(
-            gl,
-            Object.assign({}, descriptor, descriptor.layout, {
-              id: attributeName
-            })
-          );
+        const buffer = descriptor;
+        attribute = attribute || new Attribute(gl, {id: attributeName, accessor: buffer.accessor});
```

Comment:
Setting `accessor` should be covered in line 603?

Reply:
As far as I can tell the existing code only picks `type` and `divisor` from external buffers when set. 

These special cases are one of the reasons I found the `Attribute` code hard to understand/maintain.